questionnaires/superset/utils.py

```diff
@@ -85,12 +87,14 @@ def _create_dataset(self, database_id, owner_id):
         dataset_id = resp.get('id')
 
         dataset_detail = self.client.get_dataset(dataset_id)
-        columns = copy(dataset_detail.get('result', {}).get('columns'))
-        for column in columns:
-            column.pop('changed_on', None)
-            column.pop('created_on', None)
-            column.pop('type_generic', None)
-            column.pop('uuid', None)
+        columns = []
+        for column in copy(dataset_detail.get('result', {}).get('columns')):
+            if column.get('column_name') in ALLOWED_COLUMNS:
+                column.pop('changed_on', None)
+                column.pop('created_on', None)
+                column.pop('type_generic', None)
+                column.pop('uuid', None)
+                columns.append(column)
```

Comment:
This doesn't seem safe to me. It would be better to use a list comprehension with a function that extracts what we need from the response, creating a new column definition, e.g.

```
columns = [ for extract_column_definition(column) in dataset_detail.get('result', {}).get('columns') if column.get('column_name') in ALLOWED_COLUMNS ]
```

Reply:
Fixed it.